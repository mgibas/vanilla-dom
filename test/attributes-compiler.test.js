const compiler = require('../src/attributes-compiler')
let attributeUpdateCompiler = require('../src/attribute-update-compiler')
attributeUpdateCompiler.compile = jest.fn()
attributeUpdateCompiler.compile.mockReturnValue('UPDATE_COMPILER_PART')

describe('attributes-compiler', () => {
  describe('on compile', () => {
    it('simple attributes', () => {
      expect(compiler.compile('element', {foo: 'bar', bar:'foo'})).toMatchSnapshot();
    })
    it('reactive expressioon attribute', () => {
      expect(compiler.compile('element', {foo: '{{state.hey}}'})).toMatchSnapshot();
    })
    it('multiple reactive expressioons in attribute value', () => {
      expect(compiler.compile('element', {foo: '{{state.hey}} and {{state.blah}}'})).toMatchSnapshot();
    })
  })
})
