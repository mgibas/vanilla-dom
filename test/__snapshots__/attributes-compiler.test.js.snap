// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attributes-compiler on compile multiple reactive expressioons in attribute value 1`] = `
"var val = \`\${ (() => {
  try {
    return state.hey;
  } catch (err) {
    return '';
  }
})() || '' } and \${ (() => {
  try {
    return state.blah;
  } catch (err) {
    return '';
  }
})() || '' }\`;
if (val) {
  element.setAttribute('foo', val);
}
REACTIVE_COMPILER_PART;"
`;

exports[`attributes-compiler on compile reactive expressioon attribute 1`] = `
"var val = \`\${ (() => {
  try {
    return state.hey;
  } catch (err) {
    return '';
  }
})() || '' }\`;
if (val) {
  element.setAttribute('foo', val);
}
REACTIVE_COMPILER_PART;"
`;

exports[`attributes-compiler on compile simple attributes 1`] = `
"element.setAttribute('foo', \`bar\`);
element.setAttribute('bar', \`foo\`);"
`;
